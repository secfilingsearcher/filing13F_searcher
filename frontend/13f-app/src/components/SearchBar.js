import React, { useState } from 'react'
// import { Link } from 'react-router-dom'
import './SearchBar.css'
import './Button.css'
import { InputGroup, Form, Button } from 'react-bootstrap'
import { useHistory, useLocation } from 'react-router-dom'

function SearchBar () {
  const location = useLocation()
  const params = new URLSearchParams(location.search)
  const searchParam = params.get('q') || ''
  const startDateParam = params.get('startDate') || ''
  const endDateParam = params.get('endDate') || ''

  const [searchName, setSearchName] = useState(searchParam)
  const [searchStartDate, setSearchStartDate] = useState(startDateParam)
  const [searchEndDate, setSearchEndDate] = useState(endDateParam)
  const handleNameChange = event => { setSearchName(event.target.value) }
  const handleStartDateChange = event => { console.log(event.target.value); setSearchStartDate(event.target.value) }
  const handleEndDateChange = event => { setSearchEndDate(event.target.value) }
  const searchLink = `/search?q=${searchName}&start_date=${searchStartDate}&end_date=${searchEndDate}`

  const history = useHistory()
  const onSubmit = (event) => {
    event.preventDefault()
    history.push(searchLink, { replace: true })
  }

  return (
          <>
                <Form onSubmit={onSubmit}>
                    <InputGroup>
                      <Form.Control value={searchName} onChange={handleNameChange} placeholder='Company Name' />
                      <Form.Control type="date" value={searchStartDate} onChange={handleStartDateChange} />
                      <Form.Control type="date" value={searchEndDate} onChange={handleEndDateChange} />
                      <Button variant="primary" href={searchLink} type='submit' onClick={onSubmit}><i className="bi bi-search"></i></Button>
                    </InputGroup>
                </Form>
          </>)
}

export default SearchBar
