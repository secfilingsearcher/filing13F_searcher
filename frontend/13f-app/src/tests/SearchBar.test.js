import React from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import SearchBar from '../components/SearchBar'

test('SearchBar renders', ()=> {
  // August 4th, 2021
  const date = (new Date(2021, 7, 4)).toISOString().substr(0, 10)
  const component = renderer.create(
        <BrowserRouter><SearchBar startDate={date} endDate={date}/></BrowserRouter>
  )
let tree = component.toJSON()
expect(tree).toMatchSnapshot()
})