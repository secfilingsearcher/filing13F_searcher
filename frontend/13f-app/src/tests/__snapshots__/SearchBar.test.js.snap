// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders 1`] = `
<form>
  <div>
    <input
      className="search"
      onChange={[Function]}
      placeholder="Company Name"
      type="text"
      value=""
    />
    <input
      className="date"
      onChange={[Function]}
      type="date"
      value="2021-08-04"
    />
    <input
      className="date"
      onChange={[Function]}
      type="date"
      value="2021-08-04"
    />
    <a
      href="/search?q=&startDate=2021-08-04&endDate=2021-08-04"
      onClick={[Function]}
    >
      <button
        className="bar_button"
        type="submit"
      >
        SEARCH
      </button>
    </a>
  </div>
</form>
`;
